src/Analyzers/CSharp/Analyzers/SimplifyPropertyPattern/CSharpSimplifyPropertyPatternDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,83 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.CodeStyle;
+using Microsoft.CodeAnalysis.CSharp.CodeStyle;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.SimplifyPropertyPattern
+{
+    /// <summary>
+    /// Looks for code of the form:
+    /// 
+    ///     x is { a: { b: ... } }
+    ///     
+    /// and converts it to:
+    /// 
+    ///     x is { a.b: ... }
+    /// </summary>
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    internal class CSharpSimplifyPropertyPatternDiagnosticAnalyzer : AbstractBuiltInCodeStyleDiagnosticAnalyzer
+    {
+        public CSharpSimplifyPropertyPatternDiagnosticAnalyzer()
+            : base(IDEDiagnosticIds.SimplifyPropertyPatternDiagnosticId,
+                   EnforceOnBuildValues.UseLocalFunction,
+                   CSharpCodeStyleOptions.PreferSimplePropertyPattern,
+                   LanguageNames.CSharp,
+                   new LocalizableResourceString(nameof(CSharpAnalyzersResources.Property_pattern_can_be_simplified), CSharpAnalyzersResources.ResourceManager, typeof(CSharpAnalyzersResources)),
+                   new LocalizableResourceString(nameof(CSharpAnalyzersResources.Simplify_property_pattern), CSharpAnalyzersResources.ResourceManager, typeof(CSharpAnalyzersResources)))
+        {
+        }
+
+        public override DiagnosticAnalyzerCategory GetAnalyzerCategory()
+            => DiagnosticAnalyzerCategory.SemanticSpanAnalysis;
```

Comment:
Would `SyntaxTreeWithoutSemanticsAnalysis` suffice?

Reply:
unfortunately no:

```
 /// Note: an <see cref="DiagnosticAnalyzer"/> that uses this will not work properly if
/// it registers a <see cref="AnalysisContext.RegisterSyntaxNodeAction{TLanguageKindEnum}(Action{SyntaxNodeAnalysisContext}, TLanguageKindEnum[])"/> and then ends
```

Reply:
you only use that if you do RegisterSyntaxTreeAction